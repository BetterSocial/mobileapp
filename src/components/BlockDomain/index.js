/* eslint-disable no-underscore-dangle */
import * as React from 'react';
import {InteractionManager, View} from 'react-native';

import BlockDomain from '../Blocking/BlockDomain';
import ReportDomain from '../Blocking/ReportDomain';
import SpecificIssue from '../Blocking/SpecificIssue';
import blockUtils from '../../service/utils/blockUtils';
import {checkBlockDomainPage} from '../../service/domain';

class BlockDomainComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      messageReport: '',
      reportOption: []
    };

    this.refBlockDomain = React.createRef();
    this.refSpecificIssue = React.createRef();
    this.refReportDomain = React.createRef();
    this.refInteractionManager = React.createRef();

    this.__getSpecificIssue = this.__getSpecificIssue.bind(this);
    this.__onNextQuestion = this.__onNextQuestion.bind(this);
    this.__onSkipOnlyBlock = this.__onSkipOnlyBlock.bind(this);
    this.__onSelectBlock = this.__onSelectBlock.bind(this);
    this.__onBlockDomain = this.__onBlockDomain.bind(this);
  }

  componentWillUnmount() {
    if (this.refInteractionManager.current) this.refInteractionManager.current.cancel();
  }

  openBlockDomain() {
    this.refInteractionManager.current = InteractionManager.runAfterInteractions(() => {
      this.refBlockDomain.current.open();
    });
  }

  __getSpecificIssue(v) {
    this.setState({
      messageReport: v
    });
    this.refSpecificIssue.current.close();
    setTimeout(() => {
      this.__onBlockDomain();
    }, 500);
  }

  __onNextQuestion(v) {
    this.setState({
      reportOption: v
    });
    this.refReportDomain.current.close();
    this.refInteractionManager.current = InteractionManager.runAfterInteractions(() => {
      this.refSpecificIssue.current.open();
    });

    if (this.props.onReasonsSubmitted) this.props.onReasonsSubmitted(v);
  }

  __onSelectBlock(v) {
    if (v === 1) {
      this.__onBlockDomain();
    } else {
      this.refInteractionManager.current = InteractionManager.runAfterInteractions(() => {
        this.refReportDomain.current.open();
      });
    }
    this.refBlockDomain.current.close();
  }

  __onSkipOnlyBlock() {
    this.refReportDomain.current.close();
    this.refSpecificIssue.current.close();
    this.__onBlockDomain();
    if (this.props.onSkipOnlyBlock) this.props.onSkipOnlyBlock();
  }

  __onBlockDomain() {
    blockUtils.uiBlockDomain(
      this.props.domainId,
      this.state.reportOption,
      this.state.messageReport,
      this.props.screen || 'domain_screen',
      async () => {
        // getBlockedDomain(this.props.domainId).then((res) => {
        //     this.props.getValueBlock(res.data)
        // })

        const blockedDomain = await checkBlockDomainPage(this.props.domainId);
        this.props.getValueBlock(blockedDomain);
      }
    );

    if (this.props.onReportInfoSubmitted) this.props.onReportInfoSubmitted();
  }

  render() {
    return (
      <View>
        <BlockDomain
          refBlockDomain={this.refBlockDomain}
          onSelect={this.__onSelectBlock}
          domain={this.props.domain}
          onClose={() => this.props?.onCloseBlockDomain?.()}
          onBlockAndReportDomain={() => this.props?.onBlockAndReportDomain?.()}
          onBlockDomainIndefinitely={() => this.props?.onBlockDomainIndefinitely?.()}
        />
        <SpecificIssue
          refSpecificIssue={this.refSpecificIssue}
          onPress={this.__getSpecificIssue}
          onSkip={this.__onSkipOnlyBlock}
        />
        <ReportDomain
          ref={this.refReportDomain}
          onSkip={this.__onSkipOnlyBlock}
          onSelect={this.__onNextQuestion}
        />
      </View>
    );
  }
}

export default BlockDomainComponent;
