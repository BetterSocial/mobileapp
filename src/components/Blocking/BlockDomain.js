import * as React from 'react';
import RBSheet from 'react-native-raw-bottom-sheet';
import {StyleSheet, Text, View} from 'react-native';

import ItemListLarge from './ItemListLarge';
import {fonts} from '../../utils/fonts';
import {COLORS} from '../../utils/theme';

const BlockDomain = ({refBlockDomain, onSelect, domain}) => {
  const data = [
    {
      id: 1,
      icon: 'block',
      label: `Block ${domain} indefinitely`,
      desc: `You will not be able to see ${domain}’s posts. The domain’s reach will be reduced across Helio, in particular for this link.`,
      iconReght: false
    },
    {
      id: 2,
      icon: 'handcuffs',
      label: `Block & report ${domain}`,
      desc: 'This domain is fake, hacked, or involved in criminal activity. You will need to provide additional information to help us take consequences.',
      iconReght: true
    }
  ];
  return (
    <View>
      <RBSheet
        ref={refBlockDomain}
        closeOnDragDown={true}
        closeOnPressMask={true}
        customStyles={{
          container: styles.container,
          draggableIcon: styles.draggableIcon
        }}>
        <Text style={styles.title}>What do you want to do?</Text>
        <Text style={styles.desc}>
          Blocking a domain will not just remove their link from your feed, but also reduce their
          visibility for all other users.
        </Text>
        {data.map((item, index) => (
          <ItemListLarge
            key={index}
            id={item.id}
            label={item.label}
            desc={item.desc}
            iconReght={item.iconReght}
            icon={item.icon}
            onPress={() => onSelect(item.id)}
          />
        ))}
      </RBSheet>
    </View>
  );
};

export default BlockDomain;

const styles = StyleSheet.create({
  title: {
    fontFamily: fonts.inter[600],
    fontSize: 18,
    color: COLORS.white,
    marginLeft: 21
  },
  desc: {
    color: COLORS.gray500,
    fontFamily: fonts.inter[400],
    fontSize: 12,
    marginHorizontal: 21,
    marginTop: 17,
    marginBottom: 29
  },

  container: {
    height: 'auto',
    borderTopRightRadius: 20,
    borderTopLeftRadius: 20,
    backgroundColor: COLORS.almostBlack
  },
  draggableIcon: {
    backgroundColor: COLORS.gray100,
    width: 60
  }
});
