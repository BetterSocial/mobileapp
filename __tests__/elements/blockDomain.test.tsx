import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import BlockDomain from '../../src/elements/Blocking/BlockDomain';

configure({adapter: new Adapter()});
jest.useFakeTimers();
describe('component BlockDomain', () => {
  it('BlockDomain snapshot', () => {
    const myRef = {
      current: {
        open: () => {},
        close: () => {},
      },
    };
    const component = shallow(
      <BlockDomain
        domain="domain.com"
        onSelect={() => {}}
        refBlockUser={myRef}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('BlockDomain renders correctly', () => {
    const myRef = {
      current: {
        open: () => {},
        close: () => {},
      },
    };
    const tree = renderer
      .create(
        <BlockDomain
          domain="domain.com"
          onSelect={() => {}}
          refBlockUser={myRef}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
