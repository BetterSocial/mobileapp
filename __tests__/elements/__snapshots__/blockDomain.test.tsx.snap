// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component BlockDomain BlockDomain snapshot 1`] = `
<View>
  <RBSheet
    animationType="none"
    closeDuration={200}
    closeOnDragDown={true}
    closeOnPressBack={true}
    closeOnPressMask={true}
    customStyles={
      Object {
        "container": Object {
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "height": "auto",
        },
        "draggableIcon": Object {
          "backgroundColor": "#E0E0E0",
          "width": 60,
        },
      }
    }
    dragFromTopOnly={false}
    height={260}
    keyboardAvoidingViewEnabled={false}
    minClosingHeight={0}
    onClose={null}
    onOpen={null}
    openDuration={300}
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "Inter-Semi-bold",
          "fontSize": 18,
          "marginLeft": 21,
        }
      }
    >
      What do you want to do?
    </Text>
    <Text
      style={
        Object {
          "color": "#828282",
          "fontFamily": "Inter-Regular",
          "fontSize": 12,
          "marginBottom": 29,
          "marginHorizontal": 21,
          "marginTop": 17,
        }
      }
    >
      Blocking a domain will not just remove their link from your feed, but also reduce their visibility for all other users.
    </Text>
    <ItemListLarge
      desc="You will not be able to see domain.com’s posts. The domain’s reach will be reduced across BetterSocial, in particular for this link."
      icon="block"
      iconReght={false}
      id={1}
      key="1"
      label="Block domain.com indefinitely"
      onPress={[Function]}
    />
    <ItemListLarge
      desc="This domain is fake, hacked, or involved in criminal activity. You will need to provide additional information to help us take consequences."
      icon="handcuffs"
      iconReght={true}
      id={2}
      key="2"
      label="Block & report domain.com"
      onPress={[Function]}
    />
  </RBSheet>
</View>
`;
